registrations/views.py

```diff
@@ -34,6 +38,29 @@ def get_initial(self):
         return super().get_initial()
 
 
+class RegistrationConfirmOptIn(TemplateView):
+    template_name = "registrations/confirm_optin.html"
+
+    def dispatch(self, request, *args, **kwargs):
+        if (
+            "registration_details" not in request.session
+            or "msisdn" not in request.session.get("registration_details", {})
+        ):
+            return redirect(reverse_lazy("registrations:registration-details"))
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["msisdn"] = self.request.session["registration_details"]["msisdn"]
+        return context
+
+    def post(self, request, *args, **kwargs):
+        if "yes" in request.POST:
+            request.session["channel"] = "WhatsApp"
+            return redirect(reverse_lazy("registrations:confirm-clinic"))
+        return redirect(reverse_lazy("registrations:reject-optin"))
```

Comment:
Not something that necessarily needs to be changed, but I'm wondering if we should redirect to the registration-details page, maybe with a message, similar to how we do it for the "No" on the clinic confirm page?

Reply:
Hmmm :thinking: I was following the flows in the documents.
I'm not sure. Redirecting to there makes sense from the clinic confirm page because we expect them to input a new clinic code. But from here we don't expect them to input new information do we? Unless we think they might have entered the wrong msisdn?

Reply:
Yeah, as I said, not something to necessarily change, just maybe something to think about

Reply:
I do think there should at least be a button at the bottom that takes to you back to the registration-details page though. So I'll add that